app/login/_components/loginForm.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+
+import { useState } from "react";
+import { Button } from "@/components/Button/Button";
+import { Controller, useForm } from "react-hook-form";
+import { emailPattern } from "@/util/util";
+import AuthInput from "@/components/Input/AuthInput";
+import { signIn } from "next-auth/react";
+import Image from "next/image";
+import eyeIcon from "@/public/eye-on.svg";
+import eyeOffIcon from "@/public/eye_off.svg";
+import { useRouter } from "next/navigation";
+
+export interface FormValueType {
+  id: string;
+  password: string;
+  confirmPassword?: string;
+}
+
+export default function LoginForm() {
+  const [textHidden, setTextHidden] = useState(true);
+  const {
+    handleSubmit,
+    control,
+    setError,
+    formState: { isValid },
+  } = useForm<FormValueType>({ mode: "onChange" });
+  const router = useRouter();
+
+  const formAction = async (data: FormValueType) => {
+    const result = await signIn("credentials", {
+      id: data.id,
+      password: data.password,
+      redirect: false,
+    });
+    if (result?.ok) {
+      router.push("/folder");
+    } else if (result?.error) {
+      setError("id", {
+        type: "manual",
+        message: "아이디를 다시 확인해주세요!",
+      });
+      setError("password", {
+        type: "manual",
+        message: "비밀번호를 다시 확인해주세요!",
+      });
+    }
```

Comment:
@sj0724 

happy path를 먼저 고려하고 if else문을 사용하면 if else로 인해 코드의 nesting이 깊어져요. 
다음과 같이 early return을 통해 조금 더 간결하게 작성해보면 어떨까요? 

```suggestion
    if (result?.error) {
      setError("id", {
        type: "manual",
        message: "아이디를 다시 확인해주세요!",
      });
      setError("password", {
        type: "manual",
        message: "비밀번호를 다시 확인해주세요!",
      });
     return
    }
    
    router.push("/folder");
    }
```